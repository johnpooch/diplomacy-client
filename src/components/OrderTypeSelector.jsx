import React from 'react';
import styled from '@emotion/styled';

import { Button, Grid } from '../styles';
import { spacing } from '../variables';

const StyledGrid = styled(Grid)`
  margin-top: ${spacing[4]}px;
`;

const OrderSelector = (props) => {
  const { choices, onClickChoice } = props;

  const buttons = [];
  choices.forEach((choice) => {
    buttons.push(
      <Button key={choice} onClick={() => onClickChoice(choice)}>
        {choice}
      </Button>
    );
  });

  return (
    <div>
      <StyledGrid columns={buttons.length} columnGap={`${spacing[1]}px`}>
        {buttons}
      </StyledGrid>
    </div>
  );
};

export default OrderSelector;
