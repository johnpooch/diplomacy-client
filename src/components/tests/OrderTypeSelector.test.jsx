import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import OrderTypeSelector from '../OrderTypeSelector';

test('Clicking choice executes callback', () => {
  const choices = ['Choice A', 'Choice B'];
  const mockCallback = jest.fn();
  const { asFragment, getByText } = render(
    <OrderTypeSelector choices={choices} onClickChoice={mockCallback} />
  );
  const choice = getByText('Choice A');
  fireEvent.click(choice);
  expect(mockCallback.mock.calls.length).toEqual(1);
  expect(mockCallback).toHaveBeenCalledWith('Choice A');
  expect(asFragment()).toMatchSnapshot();
});
