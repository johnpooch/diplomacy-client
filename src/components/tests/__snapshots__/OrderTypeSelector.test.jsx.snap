// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking choice executes callback 1`] = `
<DocumentFragment>
  <div>
    <div
      class="css-1021laz-Grid-StyledGrid ejucsgu0"
    >
      <button
        class="css-1f7z8bo-BaseButton-Button e70lpu24"
      >
        Choice A
      </button>
      <button
        class="css-1f7z8bo-BaseButton-Button e70lpu24"
      >
        Choice B
      </button>
    </div>
  </div>
</DocumentFragment>
`;
